### PR TITLE
Add new e2e page utils and update coupon tests

### PR DESCRIPTION
**Pull request by veljkho**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Added new page utils (`applyCoupon` and `removeCoupon` ) for adding and removing coupons within cart and checkout. These require previously created coupons in order to apply them. Now, with just these page utils, you could apply them in both places without carrying if you're in the cart or the checkout flow. Also updated coupon tests.

- Perform `shopper.test.js` locally

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

N/A

### CODE REVIEW

File: tests/e2e/utils/src/page-utils.js [hunk 1]
```diff
@@ -209,6 +209,38 @@ const selectOptionInSelect2 = async ( value, selector = 'input.select2-search__f
...
+	};
+};
+
+/**
+ * Remove one coupon within cart or checkout.
+ *
+ * @returns {Promise<void>}
+ */
+const removeCoupon = async () => {
+	await expect(page).toClick('.woocommerce-remove-coupon', {text: '[Remove]'});
```
Comment (tammullen) [inline tests/e2e/utils/src/page-utils.js:239]: Could we use something like the `data-coupon` identifier here to specify the coupon `couponCode` we are removing in the event of there being multiple coupons in use.

**Review by tammullen — DISMISSED** [review]
Thanks for this update

Tests run locally and in Travis
 

```
 Checkout coupons
    ✓ allows customer to apply fixed cart coupon (1551ms)
    ✓ allows customer to apply percentage coupon (1501ms)
    ✓ allows customer to apply fixed product coupon (1548ms)
    ✓ prevents customer applying same coupon twice (1875ms)
    ✓ allows customer to apply multiple coupons (1084ms)
    ✓ restores cart total when coupons are removed (1279ms)
```

I left a comment about the removeCoupon method, but this could be covered in a new issue and not a blocker for this PR.

File: tests/e2e/utils/src/page-utils.js (hunk 1 above)
Comment (veljkho) [reply tests/e2e/utils/src/page-utils.js]: Hey, you were totally right! I just updated `removeCoupon` method so it includes the coupon name.

**Review by zhongruige — APPROVED** [review]
This tested great for me locally, thanks for adding in these utils @veljkho and updating to use `[data-coupon="'+couponCode.toLowerCase()+'"]`. Looks and runs great :)

```
  Checkout coupons
    ✓ allows customer to apply fixed cart coupon (1197ms)
    ✓ allows customer to apply percentage coupon (1195ms)
    ✓ allows customer to apply fixed product coupon (1198ms)
    ✓ prevents customer applying same coupon twice (1257ms)
    ✓ allows customer to apply multiple coupons (575ms)
    ✓ restores cart total when coupons are removed (1030ms)
```

**Pull request merged**